### PR TITLE
[TIMOB-23518] Initial Xcode 8 and iOS 10 tooling support.

### PR DESCRIPTION
**Pull request by cb1kenobi**
**JIRA:** https://jira.appcelerator.org/browse/TIMOB-23518
# DO NOT MERGE!

Still having issues with xcodebuild munging Info.plist and need to flesh out code signing a bit more.

### CODE REVIEW

File: iphone/cli/commands/_build.js [hunk 1]
```diff
@@ -1846,8 +1990,17 @@ iOSBuilder.prototype.validate = function (logger, config, cli) {
...
+				if (version.gte(this.iosSdkVersion, '10.0') && version.lt(defaultMinIosSdk, '8.0')) {
+					defaultMinIosSdk = '8.0';
+				}
+
 				this.minIosVer = this.tiapp.ios['min-ios-ver'] || defaultMinIosSdk;
-				if (version.gte(this.iosSdkVersion, '6.0') && version.lt(this.minIosVer, defaultMinIosSdk)) {
+
+				if (version.gte(this.iosSdkVersion, '10.0') && version.lt(this.minIosVer, '8.0')) {
+					logger.warn(__('The %s of the iOS section in the tiapp.xml is lower than the recommended minimum iOS version %s', 'min-ios-ver', '8.0'));
+					logger.warn(__('Consider bumping the %s to at least %s', 'min-ios-ver', '8.0'));
```
Comment (hansemannn) [inline iphone/cli/commands/_build.js]: Is that our 7.1 deprecation? ❤️

Comment (cb1kenobi) [inline iphone/cli/commands/_build.js]: If you're using Xcode 8 and building with iOS SDK 10 and you didn't explicitly set a min-ios-ver in the tiapp.xml, then yes, we default to 8.0 as the minimum.

Comment (hansemannn) [inline iphone/cli/commands/_build.js]: As we will have Xcode 8 as our recommended Xcode version in 6.0, that is a breaking change (which I highly welcome). Just that @ingo and @cheekiatng know 😊

**Comment by hansemannn** [comment]
Hey @cb1kenobi, wondering if we want this one merged for 6.0.0. Thanks!

**Comment by cb1kenobi** [comment]
@hansemannn I don't think we're going to merge this PR. I'm working on a new PR for https://jira.appcelerator.org/browse/TIMOB-23650 which takes some of the fixes from this PR.

I don't think we're going to take the team and provisioning profile changes since they aren't required by Xcode 8.

**Comment by hansemannn** [comment]
Closing for now, thanks!

**Pull request closed without merging**